.github/workflows/publish.yml

```diff
@@ -1,56 +1,100 @@
+
+      - name: Index and upload ADG
+        uses: spice-labs-inc/action-spice-labs-cli-scan@v3
+        with:
+          spice_pass: "${{ secrets.SPICE_PASS }}"
+          file_path: "${{ github.workspace }}/target"
+          tag: "${{ github.event.repository.name }}"
+
+  release_publish:
+    name: Publish to Central and GitHub Packages (tags only)
+    if: ${{ github.ref_type == 'tag' }}
     runs-on: ubuntu-24.04
-    permissions:
-      packages: write
     outputs:
       version: ${{ steps.tag.outputs.version }}
-      release_timestamp: ${{ steps.release-notification.outputs.timestamp }}
-      release_channel_id: ${{ steps.release-notification.outputs.channel_id }}
-
+      release_timestamp: ${{ steps.release_notification.outputs.timestamp }}
+      release_channel_id: ${{ steps.release_notification.outputs.channel_id }}
     steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      # Open Slack thread early, like the original workflow
       - name: Notify release
-        id: release-notification
+        id: release_notification
+        continue-on-error: true
         uses: spice-labs-inc/action-release-notification@main
         with:
           type: release
-          slack-bot-token: ${{ secrets.SLACK_BOT_TOKEN }}
-          username-mapping: ${{ secrets.GH_SLACK_USERNAME_MAPPING }}
-          github-token: ${{ github.token }}
+          slack-bot-token: "${{ secrets.SLACK_BOT_TOKEN }}"
+          username-mapping: "${{ secrets.GH_SLACK_USERNAME_MAPPING }}"
+          github-token: "${{ github.token }}"
 
-      - name: Checkout Code
-        uses: actions/checkout@v4
-
-      - name: Get version from tag (strip 'v')
+      - name: Validate and extract version from tag
         id: tag
+        shell: bash
         run: |
-          RAW_TAG="${{ github.event.release.tag_name }}"
-          VERSION="${RAW_TAG#v}"
-          echo "version=$VERSION" >> $GITHUB_OUTPUT
+          if [[ "${GITHUB_REF}" =~ ^refs/tags/v([0-9]+\.[0-9]+\.[0-9]+)$ ]]; then
+            echo "version=${BASH_REMATCH[1]}" >> "$GITHUB_OUTPUT"
+          else
+            echo "Not a semantic version tag (vX.Y.Z): ${GITHUB_REF}" >&2
+            exit 1
+          fi
 
-      - name: Validate version format (X.Y.Z)
+      - name: Ensure tag commit is on main (required for public releases)
+        id: onmain
+        shell: bash
         run: |
-          VERSION="${{ steps.tag.outputs.version }}"
-          if ! [[ "$VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-            echo "❌ Invalid version format: $VERSION"
-            exit 1
+          set -euo pipefail
+          git fetch --no-tags origin +refs/heads/main:refs/remotes/origin/main
+          if git merge-base --is-ancestor "${GITHUB_SHA}" "origin/main"; then
+            echo "on_main=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "on_main=false" >> "$GITHUB_OUTPUT"
           fi
```

Comment:
Where is this requirement coming from? The task was just to upload ADGs wasn't it?